For 517

preventicus opened the pull request.

merschel committed:
FOR-517 restruct the pcore proto structure

merschel committed:
FOR-517 rename Absolute/Differential to Uncompressed/Compressed

merschel committed:
FOR-517 Refactors Protobuf definitions for PCORE body

Updates the Protobuf definitions for the PCORE body to improve data structure and flexibility.

Renames `outer_sections_differences_ms` and `inner_sections_differences_ms` to `outer_sections_duration_ms` and `inner_sections_duration_ms` respectively, for clarity.

Restructures accelerometer and photoplethysmograph data within a unified 'Channel' message, allowing for better organization and future extensibility.

Updates the copyright year to 2025.

merschel committed:
FOR-517 Refactors data structure for PCore data.

Migrates the PCore data structure to a new format.

This change consolidates header and body information into a single 'Data' message, improving data organization and simplifying access to metadata, timestamps, and channel data.

It replaces the separate 'Header' and 'Body' messages with nested messages within the 'Data' message, providing a more structured and self-contained representation of PCore data.

merschel committed:
FOR-517 Refactors Data message in pcore.proto

Updates the Data message structure in pcore.proto to simplify data representation and optimize storage.

Replaces `UncompressedValuesContainer` and `CompressedValuesContainer` with a repeated `uint64 values` field.

Removes reserved fields and adjusts field numbers for `SensorType` oneof.

merschel committed:
Refactors data serialization format

Updates the data serialization format to improve efficiency and flexibility.

- Removes the uncompressed timestamps container, using only the compressed format.
- Changes sensor values to use a oneof field to support different data types (int and double).
- Reorders the sensor types to maintain consistency.
- Updates the class diagram in the README to reflect these changes.

merschel committed:
FOR-873 Refactors sensor data structure

Renames the `Channel` message to `Sensor` and updates related fields
to better reflect the intended representation of sensor data.

Adds support for Electrocardiogram sensor data.

Reserves unused fields in the `Sensor` message for future expansion.

merschel committed:
FOR-517 Refactors version definition in pcore

Moves the Version message definition from the separate pcore_version.proto file into the pcore.proto file.

This simplifies the project structure by removing an unnecessary file and improves code maintainability by centralizing related definitions.

merschel committed:
FOR-517 Updates pcore documentation with sensor data details

Refines the pcore documentation to include specifics
on supported sensor types (PPG, ACC, ECG) and their data
representation.

Explains the data storage strategy, emphasizing the use of
compressed timestamps and delta encoding for integer values,
enhancing clarity and understanding of the pcore data format.

merschel committed:
FOR-517 Describes timestamp serialization algorithm

Adds a detailed description of the timestamp serialization algorithm,
including pseudocode for finding section indices and the serialization
logic itself. It also covers integer and double value serialization
and deserialization methods.

piperdiana commented on README.md:
```diff
@@ -9,26 +9,42 @@
+	sensor are shared and compressed using a structure called CompressedTimestampsContainer. 
+	This container represents the measurement timeline by dividing it into 
+	sections of constant time intervals. The compressed representation stores 
+	the initial timestamp, per-section time deltas, and the number of data points per section.
```
hier fehlt die outer-section time deltas

merschel replied:
done

piperdiana commented on README.md:
```diff
@@ -9,26 +9,42 @@
+	sections of constant time intervals. The compressed representation stores 
+	the initial timestamp, per-section time deltas, and the number of data points per section.
 
 	For data compression purposes, measurement values are stored in block form:
```
die sind aber nicht mehr in blocks sonder sections oder?

merschel replied:
done

merschel committed:
FOR-517 Clarifies pcore data compression description

Updates the README to use the term "section" instead of "block"
for consistency in describing pcore data compression.
This improves the clarity and understanding of the data format.

merschel committed:
FOR-517 Clarifies compressed timestamps structure

Updates the README to accurately describe how timestamps are compressed.

It now includes information about both intra-section and inter-section time deltas.

merschel committed:
FOR-517 Updates documentation for data storage

Clarifies how measurement values are stored to improve
documentation clarity. It focuses on detailing the storage
method used for data compression.
